tests/tls_e2e/host/host.cpp

```diff
@@ -237,7 +250,12 @@ int run_test_with_config(tls_test_configs_t* test_configs)
     {
         // Release lock on scope exit
         std::unique_lock<std::mutex> server_lock(g_server_mutex);
-        g_server_cond.wait(server_lock, [] { return g_server_condition; });
+        g_server_cond.wait(
+            server_lock, [] { return g_server_initialization_done; });
+        if (!g_server_initialization_success)
+        {
+            return g_server_thread_exit_code;
```

Comment:
Consider adding an `OE_TRACE_INFO/ERROR()` here to log the server failure; it can be confusing since this only used to return the `g_client_thread_exit_code` to differentiate between the two without some disambiguation.